modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -424,6 +426,67 @@ def validate_confinements!
           end
         end
       end
+
+      def validate_form_526_direct_deposit!
+        direct_deposit = form_attributes['directDeposit']
+        return if direct_deposit.blank?
+
+        direct_deposit['noAccount'] == true ? validate_no_account! : validate_account_values!
+      end
+
+      def validate_no_account! # rubocop:disable Metrics/MethodLength
+        acc_vals = form_attributes['directDeposit']
+
+        if acc_vals['accountType'].present?
+          raise ::Common::Exceptions::UnprocessableEntity.new(
+            detail: 'If the claimant has no account the account type must be left unchecked.'
+          )
+        end
+
+        if acc_vals['accountNumber'].present?
+          raise ::Common::Exceptions::UnprocessableEntity.new(
+            detail: 'If the claimant has no account the account number field must be left empty.'
+          )
+        end
+
+        if acc_vals['routingNumber'].present?
+          raise ::Common::Exceptions::UnprocessableEntity.new(
+            detail: 'If the claimant has no account the routing number field must be left empty.'
+          )
+        end
+
+        if acc_vals['financialInstitutionName'].present?
+          raise ::Common::Exceptions::UnprocessableEntity.new(
+            detail: 'If the claimant has no account the financial institution field must be left empty.'
+          )
```

Comment:
should we break the raise in to a method and pass the differing param values here? be a little minimized. something like the below (these names are just rough examples, don't think we use them specifically)

```
def account_invalid(param_name)
  raise ::Commo....... (detail: "If the claimant has no account the #{param_name} field must be left empty")
```

Adds a bit of consistency as well. Then we can just do
```
account_invalid('account number') if acc_vals['accountNumber'].present?
```

for each of the checks here. Not really a must, but it would remove the need for the rubocop MethodLength line as well.

Reply:
Yep, good idea